osu.Game/Screens/Play/MasterGameplayClockContainer.cs

```diff
@@ -195,15 +198,12 @@ private void addSourceClockAdjustments()
             if (speedAdjustmentsApplied)
                 return;
 
-            if (SourceClock is not Track track)
-                return;
+            musicController.ResetTrackAdjustments();
```

Comment:
Feels weird to use `MusicController.ResetTrackAdjustments` when already having access over the track that is to be adjusted, i.e. could just use `track.ResetSpeedAdjustments` or `track.RemoveAllAdjustments`.

Reply:
Gonna keep this one as it is for now to not pile up more changes in this PR.